Documentation

dgrassellyb opened the pull request.

bms63 reviewed (commented):
ummm....whoa this is thorough!!!!!!!  I will review tomorrow.  `1.0` got me a little busy

manciniedoardo commented on vignettes/common_structure.Rmd:
````diff
@@ -0,0 +1,82 @@
+    image: "ghcr.io/pharmaverse/admiralci-${{ inputs.r-version }}:latest"
+```
+
+(More more details on docker images creation process and how they're used, please read [Docker images and codespaces](docker_and_codespaces.html)) section.
````
typo: "More more" instead of "For more"

dgrassellyb replied:
```suggestion
(For more details on docker images creation process and how they're used, please read [Docker images and codespaces](docker_and_codespaces.html)) section.
```

manciniedoardo commented on vignettes/common_structure.Rmd:
```diff
@@ -0,0 +1,82 @@
+
+## Staged dependencies
+
+The staged dependencies action allow to install every necessary upstream dependencies for `admiral` package, suche as :
```
typo: "dependencies" instead of "dependency"

manciniedoardo replied:
also "suche" instead of "such"

manciniedoardo replied:
also missing full stops.

dgrassellyb replied:
```suggestion
The staged dependencies action allow to install every necessary upstream dependencies for `admiral` package, such as :
```

manciniedoardo commented on vignettes/common_structure.Rmd:
````diff
@@ -0,0 +1,82 @@
+        GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+```
+
+Note : the user can specify some parameters depending on the use case :
````
The bullet pointing is not rendering properly here:
![image](https://github.com/pharmaverse/admiralci/assets/53403957/676e0119-7eb5-4456-a62c-281c5ae23892)

manciniedoardo commented on vignettes/cran-status.Rmd:
```diff
@@ -0,0 +1,59 @@
+# CRAN Status Workflow
+
+This GitHub Actions workflow checks the CRAN status of an R package. It can create an issue if specified statuses ('NOTE', 'WARN', 'ERROR') are reported on the check report.
+Note : this workflow is using [insightEngineering Cran Status Monitor](https://github.com/insightsengineering/cran-status-monitor) action.
```
typo: "insightEngineering" instead of "insightsEngineering"

dgrassellyb replied:
```suggestion
Note : this workflow is using [insightsEngineering Cran Status Monitor](https://github.com/insightsengineering/cran-status-monitor) action.
```

manciniedoardo commented on vignettes/docker_and_codespaces.Rmd:
```diff
@@ -24,6 +24,8 @@ This article will go over the following:
+
 ## What are codespaces and how to use them
 
 ### What are codespaces, and why are we using it for Admiral ?
```
I would always write "Admiral" or "admiral" as `{admiral}` for consistency (across all pages)

manciniedoardo replied:
typo below: "contributors" instead of "contributor"

manciniedoardo commented on vignettes/get_started.Rmd:
```diff
@@ -0,0 +1,101 @@
+
+Notes :
+
+- Some workflows are using github actions from [InsightEngineering](https://github.com/insightsengineering/) group.
```
typo: "InsightEngineering" instead of "InsightsEngineering"

dgrassellyb replied:
```suggestion
- Some workflows are using github actions from [InsightsEngineering](https://github.com/insightsengineering/) group.
```

dgrassellyb commented:
I also made this cheatsheet (However some links need the documentation to be deployed) 
But maybe still a bit complex to read ! 

[Template cheatsheet.pdf](https://github.com/pharmaverse/admiralci/files/13578938/Template.cheatsheet.pdf)

manciniedoardo commented on vignettes/get_started.Rmd:
```diff
@@ -0,0 +1,101 @@
+
+# Available workflows
+
+## Workflows trigger by Admiral MR (feature branch to main branch)
```
Typo: "trigger" instead of "triggered" (across other headings too)

dgrassellyb replied:
```suggestion
## Workflows triggered by Admiral MR (feature branch to main branch)
```

manciniedoardo commented on vignettes/get_started.Rmd:
```diff
@@ -0,0 +1,101 @@
+## Workflows to manage dependencies change (`renv.lock` files changes)
+
+Before talking about workflows, let's first explain how dependencies are handled in `Admiral` package.
+The developpers are using `renv` framework, to be able to use the same versions between developpers.
```
Typo: "developpers" instead of "developers"

dgrassellyb replied:
```suggestion
The developers are using `renv` framework, to be able to use the same versions between developpers.
```

manciniedoardo commented on vignettes/get_started.Rmd:
```diff
@@ -0,0 +1,101 @@
+This function is defined to determine the dependencies for the renv (R package manager) snapshot.
+It looks for the "admiral" DESCRIPTION file and extracts its dependencies, looking for suggested packages in the current project.
+
+- Then there is a step to look for current R version on a admiral devlopper environment, and load the proper R profile (use of the associate `renv.lock` file)
```
typo: "devlopper" instead of "developer"

dgrassellyb replied:
```suggestion
- Then there is a step to look for current R version on a admiral developer environment, and load the proper R profile (use of the associate `renv.lock` file)
```

manciniedoardo commented on vignettes/get_started.Rmd:
```diff
@@ -0,0 +1,101 @@
+
+- `.get_dependencies` function :
+
+This function is defined to determine the dependencies for the renv (R package manager) snapshot.
```
`{renv}`

dgrassellyb replied:
```suggestion
This function is defined to determine the dependencies for the {renv} (R package manager) snapshot.
```

manciniedoardo commented on vignettes/lock_and_prop.Rmd:
```diff
@@ -60,7 +60,7 @@ The process is implemented to generate `renv.lock` file automatically. It will b
 
 **Note:** In `[1]` you will see that one reviewer has a special shield symbol next to their name.  This person is a codeowner of the `common.yml` and is alerted whenever a change is made to this file.  You can read more about codeowners on [About Code Owners](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) in G ...
 
-Note that this propagate process also propagate .devcontainer files which are configuration files for codespaces (see [Codespaces](codespaces.Rmd))
```
In general should we add something about the fact we plan to remove this? And what we plan to switch it for?

manciniedoardo replied:
later down, there is also a missing link for admiralvaccine

manciniedoardo replied:
(also no "s" in admiralvaccine)

manciniedoardo commented on vignettes/man-pages.Rmd:
```diff
@@ -0,0 +1,59 @@
+- **Concurrency**: Man page jobs are organized by the pull request number or branch.
+- **Cancel-in-progress**: Allows canceling in-progress man page jobs.
+
+## Job: `lint`
```
is "lint" here incorrect?

manciniedoardo commented on vignettes/pkgdown.Rmd:
```diff
@@ -0,0 +1,39 @@
+# pkgdown Workflow
+
+This GitHub Actions workflow automates the generation and deployment of documentation for an R package using `pkgdown`. The workflow supports multiple versions and offers flexibility in customization.
+This workflow use rmarkdown to be able to generate html documentation from `Rmd` files (vignettes folder). This documentation is then stored in `gh-pages` branch (an orphan branch) and published.
```
typo: "use" instead of "uses"

dgrassellyb replied:
```suggestion
This workflow uses rmarkdown to be able to generate html documentation from `Rmd` files (vignettes folder). This documentation is then stored in `gh-pages` branch (an orphan branch) and published.
```

manciniedoardo commented on vignettes/pkgdown.Rmd:
```diff
@@ -0,0 +1,39 @@
+- `skip-multiversion-docs`: Skip the creation of multi-version docs. Default: false. If set to True, a subfolder will be created under `gh-pages` (with the current github reference (branch/tag) name)
+- `multiversion-docs-landing-page`: Ref to use for the multiversion docs landing page. Default: main.
+- `latest-tag-alt-name`: An alternate name to use for 'latest-tag' for multiversion pkgdown docs. Default: "".
+- `branches-or-tags-to-list`: Regular expression defining branches or tags listed under the 'Versions' dropdown menu. Default: ^main$|^devel$|^pre-release$|^latest-tag$|^cran-release$|^develop$|^v([0-9]+\\.)?([0-9]+\\.)?([0-9]+)$.
```
Something funky is happening here 😆 
![image](https://github.com/pharmaverse/admiralci/assets/53403957/ea3cf032-f40e-4108-887f-56c6104e68f8)

dgrassellyb replied:
```suggestion
- `branches-or-tags-to-list`: Regular expression defining branches or tags listed under the 'Versions' dropdown menu. Default: `^main$|^devel$|^pre-release$|^latest-tag$|^cran-release$|^develop$|^v([0-9]+\\.)?([0-9]+\\.)?([0-9]+)$`.
```

manciniedoardo commented on vignettes/push-docker.Rmd:
```diff
@@ -0,0 +1,74 @@
+# Deploy Docker Image Workflow
+
+This GitHub Actions workflow automates the deployment of Docker images based on changes in the `renv.lock` file within the `renv/profiles` directory. It is designed to run on pushes to the `main` branch, manual triggers, and on a schedule (every day at midnight UTC on Wednesdays and Sundays).
+Note : this workflow is using [InsightEngineering R image creator action](https://github.com/insightsengineering/r-image-creator).
```
typo: "InsightEngineering" instead of "InsightsEngineering"

dgrassellyb replied:
```suggestion
Note : this workflow is using [InsightsEngineering R image creator action](https://github.com/insightsengineering/r-image-creator).
```

manciniedoardo commented on vignettes/r-cmd-checks.Rmd:
```diff
@@ -0,0 +1,70 @@
+
+### `R-CMD-check` Job
+
+This job runs R CMD checks on different R versions, leveraging specified CRAN repositories. The job is conditioned not to run if the commit message contains '[skip r-cmd-check]'. The checks are performed in parallel for each specified R version, using [Matrix Strategy](#matrix)
```
Missing full stop at the end.

dgrassellyb replied:
```suggestion
This job runs R CMD checks on different R versions, leveraging specified CRAN repositories. The job is conditioned not to run if the commit message contains '[skip r-cmd-check]'. The checks are performed in parallel for each specified R version, using [Matrix Strategy](#matrix).
```

manciniedoardo commented on vignettes/readme-render.Rmd:
```diff
@@ -0,0 +1,47 @@
+
+# Render README Workflow
+
+This GitHub Actions workflow renders the README from R Markdown (`README.Rmd`) to Markdown (`README.md`). Additionally, it can format the generated Markdown using a formatter. The workflow is triggered manually (`workflow_dispatch`) and can also be invoked as a workflow call (`workflow_call`). 
```
Could you specify why this is required?

manciniedoardo commented on vignettes/spellchecks.Rmd:
```diff
@@ -0,0 +1,48 @@
+
+# Spelling Workflow
+
+This GitHub Actions workflow performs spellchecking on the specified R files. The workflow is triggered manually (`workflow_dispatch`) and can also be invoked as a workflow call (`workflow_call`).
```
Mention somewhere the exclusion wordlist here?

manciniedoardo commented:
Can we add a hex to the front page too? smth like this?
![Uploading admiralci_logo.png…]()

manciniedoardo commented:
![admiralci_logo](https://github.com/pharmaverse/admiralci/assets/53403957/ec095cec-b3d1-4dbb-a9ef-dfde5b0b1fb8)

dgrassellyb commented:
> Thanks @dgrassellyb for this, this is fantastic!!
> 
> I left a few specific comments for typos and requests for clarification. I also have some comments about the front page README:
> 
> * Many heading links in the readme don't work?
> * Full stop missing at the end of renv propagation section
> * Boilerplate or boilderplate? Could be a typo.
> * docker image build: typo in "each R version" (no "s" at the end)
> 
> I also have some general comments:
> 
> * Within the navbar, for the User Guides section I would put a subsection titled "Basics" with the "Get Started", "Common structure of the workflows" and "Codespaces" sections. Then I would put another subsection titled "Invididual workflows" with all the other pages.
> * I would retitile "common structure" to "Common strucure of the workflows" and put it at the top of the articles.
> * There's also another "spell checks page" in the Articles drop down. Not sure what the difference is between the two?

Thanks a lot for your feedbacks, I will review this ! :)

bms63 commented:
> I also made this cheatsheet (However some links need the documentation to be deployed) But maybe still a bit complex to read !
> 
> [Template cheatsheet.pdf](https://github.com/pharmaverse/admiralci/files/13578938/Template.cheatsheet.pdf)

above and beyond!!!  This cheatsheet is so cool

bundfussr commented on vignettes/renv-lock.Rmd:
```diff
@@ -0,0 +1,51 @@
+
+- `contents: write`: Write permissions for repository contents.
+
+Note : This action is quite hard to maintain, there are indeed some automation parts, such as `get_dependencies` function to configure `renv` snapshot with the `DESCRIPTION` file (taking also Suggests deps).
```
```suggestion
Note : This action is quite hard to maintain, there are indeed some automation parts, such as `.get_dependencies` function to configure `renv` snapshot with the `DESCRIPTION` file (taking also Suggests deps).
```

bundfussr commented on vignettes/spellchecks.Rmd:
```diff
@@ -0,0 +1,48 @@
+
+## Workflow Structure
+
+The workflow consists of a single job: `roxygen`.
```
Why is the job called `roxygen`?

dgrassellyb replied:
(indeed this is an error .. I'll update this in my next commit)

bundfussr commented on vignettes/validation.Rmd:
```diff
@@ -0,0 +1,52 @@
+---
+title: "Spell Checks"
+output: 
+  rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Spell Checks}
```
```suggestion
title: "Validation Report"
output: 
  rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Validation Report}
```

bundfussr reviewed (commented):
Looks great overall.

Maybe we should structure the "Articles" menu, e.g., separate sections for workflows used in all admiral packages and workflows used in admiralci only.

dgrassellyb commented on vignettes/common_structure.Rmd:
```diff
@@ -0,0 +1,82 @@
+## Use of `admiralci` docker images
+
+Once the `r-version` is picked, the worfklows will run inside a docker container instance. Indeed, `admiralci` contains an action `push-docker-image`, to be able to build `rocker:rstudio` images
+for each available versions of `R` (versions specified in `renv/profiles`, see [renv profiles conf](get_started.html#renv-profiles-configuration-handling-several-r-versions)). 
```
```suggestion
for each available version of `R` (versions specified in `renv/profiles`, see [renv profiles conf](get_started.html#renv-profiles-configuration-handling-several-r-versions)). 
```

dgrassellyb commented on vignettes/common_structure.Rmd:
```diff
@@ -0,0 +1,82 @@
+for each available versions of `R` (versions specified in `renv/profiles`, see [renv profiles conf](get_started.html#renv-profiles-configuration-handling-several-r-versions)). 
+At the building step of these docker images, every dependencies specified in the `renv.lock` files are installed.
+A total of 4 images are created :
+- `admiralci-4.1`
+- `admiralci-4.2`
+- `admiralci-4.3`
+- `admiralci-devel`
```
```suggestion

- `admiralci-4.1`
- `admiralci-4.2`
- `admiralci-4.3`
- `admiralci-devel`
```

dgrassellyb commented on vignettes/common_structure.Rmd:
```diff
@@ -0,0 +1,82 @@
+- `admiralci-4.3`
+- `admiralci-devel`
+
+These images and store them in admiralci ghcr (Github Container Registry - Place to store docker images), and are then reused to execute the workflows in docker container instances like this :
```
```suggestion
These images are stored in admiralci ghcr (Github Container Registry - Place to store docker images), and are then reused to execute the workflows in docker container instances like this :
```

dgrassellyb commented on vignettes/common_structure.Rmd:
````diff
@@ -0,0 +1,82 @@
+```
+
+Note : the user can specify some parameters depending on the use case :
+- run-system-dependencies : install or not sys dependencies
+- renv-restore : restore or not renv.lock dependencies
+(and many other options you can find on Staged Dependencies Action documentation).
````
```suggestion

- run-system-dependencies : install or not sys dependencies
- renv-restore : restore or not renv.lock dependencies
(and many other options you can find on Staged Dependencies Action documentation).
```

dgrassellyb commented:
@bms63 / @manciniedoardo / @cicdguy / @bundfussr : 
I am currently updating the doc (quite lot of changes since we removed renv !) - I just have a question about codespaces, our actual config is quite outdated (since we built it from renv dependencies) - I could work on this, but while it's not ready I was just thinking to tag the codespace section as "WIP article", what do you think ?
 
I guess codespaces feature is a bit low priority right ? (I am not sure if devs are really using it on admiral but let me know if I am wrong ! 🙊 )

dgrassellyb commented:
> @bms63 / @manciniedoardo / @cicdguy / @bundfussr : I am currently updating the doc (quite lot of changes since we removed renv !) - I just have a question about codespaces, our actual config is quite outdated (since we built it from renv dependencies) - I could work on this, but while it's not ready I was just thinking to tag the codespace section as "WIP article", what do you think ?
> 
> I guess codespaces feature is a bit low priority right ? (I am not sure if devs are really using it on admiral but let me know if I am wrong ! 🙊 )



> I'd vote to keep at as a "WIP article" and get back to it later.

Ok let's do this ! it's ready to be merge now (I also pushed a fix in `pkgdown` to be able to push non-multi version documentation - because I guess we don't need multi version for admiralci doc !). 
I also updated the cheatsheet here : 
https://github.com/pharmaverse/admiralci/blob/documentation/vignettes/assets/cheatsheet.png

bms63 commented:
Shall we just merge and publish this and do the review in an issue?  Might be a little easier if we don't have to pull down and render the site??

dgrassellyb commented:
> Shall we just merge and publish this and do the review in an issue? Might be a little easier if we don't have to pull down and render the site??

I'll fork it for rendering and send you the link

dgrassellyb commented:
Pushed again some other fixs, here is the rendered doc from my forked repo : https://dgrassellyb.github.io/admiralci/ 
However .. I have an issue in r cmd checks for this PR - not sure to understand where it's coming from : 
```
❯ checking CRAN incoming feasibility ... [1s/13s] NOTE
  Maintainer: ‘Open Source <admiralci@pharmaverse.com>’
  
  Found the following (possibly) invalid URLs:
    URL: 
      From: inst/doc/push-docker.html
      Message: Empty URL

0 errors ✔ | 0 warnings ✔ | 1 note ✖
```

I even updated the `.Buildignore` with `^inst/doc\.html$` but still get this

manciniedoardo commented on README.md:
```diff
@@ -1,229 +1,151 @@
-reusable via the
-[`workflow_call`](https://docs.github.com/en/actions/using-workflows/reusing-workflows)
-GitHub Actions event.
+reusable via the \[`workflow_call`]\[workflow\_call] GitHub Actions
```
@dgrassellyb is it that I don't understand what this syntax is saying or is it a typo? Don't be afraid to tell me if it's the former 😄

dgrassellyb replied:
Indeed since to comes from .Rmd -> .md job, I have this in the .Rmd : 

 reusable via the [`workflow_call`][workflow_call] GitHub Actions event.

manciniedoardo replied:
ok, thanks!

bms63 commented:
Can we minimize this image please 
![image](https://github.com/pharmaverse/admiralci/assets/10111024/0c6424d9-874f-4310-9c45-3c7ce60d3d73)

bms63 commented:
Can we change the tab name to CI/CD Workflows
![image](https://github.com/pharmaverse/admiralci/assets/10111024/bbe075c4-3c58-4e1f-8878-6402a01cf807)

When I get a dozen tabs up be nice if I new which admiral to select??

bms63 commented:
What about removing these links and just listing out our potential workflows?  The one I clicked go to 404.  

![image](https://github.com/pharmaverse/admiralci/assets/10111024/cfdc5c42-fb38-4306-840f-f5a8d04f1b93)

Can we say why this is a bad practice?  can't get latest and greatest updates - encourage you to make issues to enchance our workflows.  

![image](https://github.com/pharmaverse/admiralci/assets/10111024/47c40548-d7ae-4ad5-9269-76db3e765ace)

Could we give some context on what these bullet points are leading us to?

![image](https://github.com/pharmaverse/admiralci/assets/10111024/ec22d268-4f5c-4efc-ac25-49c2c506a0f0)

bms63 commented:
Possible to embed cheat sheet like we did for admiral? 
![image](https://github.com/pharmaverse/admiralci/assets/10111024/a89eebfe-65c0-4afd-9cfa-cb46565ebaa7)

Shall we remove `devel` references 
![image](https://github.com/pharmaverse/admiralci/assets/10111024/916f2485-8e28-4c3d-9560-5c7e1ee2b5da)


Many thanks again!!  Could be a really good reference for discussion for upcoming admiralpeds @rossfarrugia

dgrassellyb commented:
> Possible to embed cheat sheet like we did for admiral? ![image](https://private-user-images.githubusercontent.com/10111024/303745979-a89eebfe-65c0-4afd-9cfa-cb46565ebaa7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc1MDQ4NTIsIm5iZiI6MTcwNzUwNDU1MiwicGF0aCI6Ii8xMDExMTAyNC8zMDM3NDU5NzktYTg5ZWViZmUtNjVjMC00YWZkLTljZmEtY2I0NjU2NWViYWE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjA5VDE4NDkxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ5MzhkZjIxNDRjZmI1YjQ0NzM0MjY0NGIyOGY3ZTE1NjA4YWQ5OTVjOWJmZTc0YzdiNjJmNzVkNjFiOTc2NmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.c_Y9-_wMudsV7Lc_g53hOpl8008PYJW9KhlDZXA_gs0)
> 
> Shall we remove `devel` references ![image](https://private-user-images.githubusercontent.com/10111024/303746398-916f2485-8e28-4c3d-9560-5c7e1ee2b5da.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc1MDQ4NTIsIm5iZiI6MTcwNzUwNDU1MiwicGF0aCI6Ii8xMDExMTAyNC8zMDM3NDYzOTgtOTE2ZjI0ODUtOGUyOC00YzNkLTk1NjAtNWM3ZTFlZTJiNWRhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjA5VDE4NDkxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRmMWMyNjVjZDFjNzhmODExNDg5NzJkYWU5MjZlODBjYTRhNWY5NDc3NWY2MWJmYTg0NWEyYjQ5YWE1MDg0ZjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._mC-PdpFB-fpjuJCvOYteyICsfm7iCCOsz1mqJmL5ps)
> 
> Many thanks again!! Could be a really good reference for discussion for upcoming admiralpeds @rossfarrugia

Thanks Ben I will apply some changes as you suggested (and also for your other comments above ;) )

dgrassellyb commented:
> What about removing these links and just listing out our potential workflows? The one I clicked go to 404.
> 
> ![image](https://private-user-images.githubusercontent.com/10111024/303745507-cfdc5c42-fb38-4306-840f-f5a8d04f1b93.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc1MDQ4NTIsIm5iZiI6MTcwNzUwNDU1MiwicGF0aCI6Ii8xMDExMTAyNC8zMDM3NDU1MDctY2ZkYzVjNDItZmIzOC00MzA2LTg0MGYtZjVhOGQwNGYxYjkzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjA5VDE4NDkxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc0OGVmNWNjZTU2NDE1OTg1Njk3ZWFmMzBiZGE2MTJkNmYyMTdiYzc4ZmVmYzY3YTJkMjdiMDUxNjFhYTlhZDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.2KEvixxa5X7Utw7o5_gWM6Ei6mpDaozcYq8OMhZ_Kho)
> 
> Can we say why this is a bad practice? can't get latest and greatest updates - encourage you to make issues to enchance our workflows.
> 
> ![image](https://private-user-images.githubusercontent.com/10111024/303745538-47c40548-d7ae-4ad5-9269-76db3e765ace.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc1MDQ4NTIsIm5iZiI6MTcwNzUwNDU1MiwicGF0aCI6Ii8xMDExMTAyNC8zMDM3NDU1MzgtNDdjNDA1NDgtZDdhZS00YWQ1LTkyNjktNzZkYjNlNzY1YWNlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjA5VDE4NDkxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAxZTExMGExYTBkZDZkZjJkNDEzNmM1MGQ3YThhZjM0NGZmNDQ1ZmFlMTkxNDk0Y2Y4MDdhYjRmNDFmN2I3ZWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qbpzLT10H2M8SdD8_d4u7ZJxOXQ-xLE-XfTrxE9_6cc)
> 
> Could we give some context on what these bullet points are leading us to?
> 
> ![image](https://private-user-images.githubusercontent.com/10111024/303745688-ec22d268-4f5c-4efc-ac25-49c2c506a0f0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc1MDQ4NTIsIm5iZiI6MTcwNzUwNDU1MiwicGF0aCI6Ii8xMDExMTAyNC8zMDM3NDU2ODgtZWMyMmQyNjgtNGY1Yy00ZWZjLWFjMjUtNDljMmM1MDZhMGYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjA5VDE4NDkxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY1NGIyNmM5YTMwY2QwZTc1YTZjZWFmNTc1MWEzOTg3MTk1N2I1NjRlMmI2YTBmNTNhOGJlMDM1YTA3YWU4ZjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Pim9Y4pb_UbdhjVl4Tk8A0xSwmeiAtzTHhtVYOOZLZo)

For the links it's not working because it's coming from my fork (if you replace on links pharmaverse by dgrassellyb it will work) - we won't have issues with links once it will be deployed in admiralci repo (non-forked one !) - I think it's nice to keep it (it's like a table of contents with redirections to detail documentation + workflows code)
